### ISSUE TITLE
About Dialog box is not showing correct build-id

### ISSUE THREAD

**Issue by sravanlakkimsetti** [eclipse-platform/eclipse.platform#59]
About dialog is showing old build id instead of the latest build id. Splash screen shows the correct build id

![image](https://user-images.githubusercontent.com/77106773/180402010-0bcf418b-a75c-4a62-a3ed-44a4b0ca4014.png)
the build id should have been I20220722-0020

**Comment by iloveeclipse** [comment]
@HannesWell : this is a regression from https://github.com/eclipse-platform/eclipse.platform/pull/53, more specific from this commit : 0a954de3c69870625b1b33c605c35bb036f9b326

Whatever the tycho does now, it doesn't properly re-generate **org.eclipse.sdk/about.mappings** file which is now **always** has this content:
```
0=I20220718-1800
1=2022-06
2=4.25
```

The [checked in template about.mappings](https://github.com/eclipse-platform/eclipse.platform/blob/master/platform/org.eclipse.sdk/about.mappings) has this content:
```
0=${buildId}
1=${releaseName}
2=${releaseNumberSDK}
```

I assume maven doesn't properly compute the values now.
This must be fixed.

**Comment by HannesWell** [comment]
Yes, that should be fixed. Just I had quick look at it, but could not find out where the `buildId` property is coming from exactly.
I will have a deeper look tomorrow. One guess would be that `buildTimestamp` has to be used instead, but I'm not yet 100% certain and the question is as well, why it worked before the other way.

**Comment by iloveeclipse** [comment]
I can understand where I20220718-1800 is coming from (last change on SDK bundle).
I assume tycho simply doesn't recompute buildId because of no changes since.
But what I can't understand is: why releaseName is 2022-06?

**Comment by laeubi** [comment]
The values are defined [in the parent](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/0288585a3dc80f4be45bfb4c2b29ed6977dc4912/eclipse-platform-parent/pom.xml#L43) and [later replaced by maven](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/0288585a3dc80f4be45bfb4c2b29ed6977dc4912/eclipse-platform-parent/pom.xml#L969), so Tycho has not much to do here...

**Comment by iloveeclipse** [comment]
> But what I can't understand is: why releaseName is 2022-06?

I will fix that in a moment, it is unrelated to the build ID problem here.

**Comment by iloveeclipse** [comment]
> > But what I can't understand is: why releaseName is 2022-06?
> 
> I will fix that in a moment, it is unrelated to the build ID problem here.

**No, it is not unrelated**.

Even after https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/417 merged, about dialog is wrong on new SDK build, for **both** build id and release name in https://download.eclipse.org/eclipse/downloads/drops4/I20220723-0730 !

Maven build **replaces** built version with cached version of **org.eclipse.sdk_4.25.0.v20220718-1722** bundle now!
https://ci.eclipse.org/releng/view/Builds/job/I-build-4.25/100/consoleText
```
[2022-07-23T12:48:20.206Z] [INFO] Signing jar: /home/jenkins/agent/workspace/I-build-4.25/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform/platform/org.eclipse.sdk/target/org.eclipse.sdk-4.25.0-SNAPSHOT.jar
[2022-07-23T12:48:21.137Z] [INFO] 
[2022-07-23T12:48:21.137Z] [INFO] --- tycho-p2-plugin:3.0.0-SNAPSHOT:p2-metadata (p2-metadata) @ org.eclipse.sdk ---
[2022-07-23T12:48:21.137Z] [INFO] Artifact comparison detailed log directory /home/jenkins/agent/workspace/I-build-4.25/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform/platform/org.eclipse.sdk/target/artifactcomparison
[2022-07-23T12:48:21.137Z] [WARNING] MavenProject: eclipse.platform:org.eclipse.sdk:4.25.0-SNAPSHOT @ /home/jenkins/agent/workspace/I-build-4.25/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform/platform/org.eclipse.sdk/.polyglot.META-INF: baseline and build artifacts have same version but different contents
[2022-07-23T12:48:21.137Z]    no-classifier: different
[2022-07-23T12:48:21.137Z]       about.mappings: properties files differ
[2022-07-23T12:48:21.137Z]          0: baseline='I20220718-1800' != reactor='I20220723-0730'
[2022-07-23T12:48:21.137Z]          1: baseline='2022-06' != reactor='2022-09'
[2022-07-23T12:48:21.137Z] 
[2022-07-23T12:48:21.137Z] [INFO] MavenProject: eclipse.platform:org.eclipse.sdk:4.25.0-SNAPSHOT @ /home/jenkins/agent/workspace/I-build-4.25/eclipse.platform.releng.aggregator/eclipse.platform.releng.aggregator/cje-production/gitCache/eclipse.platform.releng.aggregator/eclipse.platform/platform/org.eclipse.sdk/.polyglot.META-INF
[2022-07-23T12:48:21.137Z]     The main artifact has been replaced with the baseline version.
[2022-07-23T12:48:21.137Z] 
```

![image](https://user-images.githubusercontent.com/964108/180609844-3b7cf246-2f4c-4f24-92ce-ba9badcd5bec.png)

**Comment by iloveeclipse** [comment]
@HannesWell : can't we simple restore the pom & property files for platform/org.eclipse.sdk & platform/org.eclipse.platform ? See #60

**Comment by iloveeclipse** [comment]
Works again after revert, see https://download.eclipse.org/eclipse/downloads/drops4/I20220723-1800/

**Comment by HannesWell** [comment]
Thank you Andrey for partially reverting.
I investigated the issue and the reason for this issue is that in the `eclipse-platform-parent/pom.xml` the `format` configuration-property of the `tycho-packaging-plugin` is set to [`<format>${qualifier.format}</format>`](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/0288585a3dc80f4be45bfb4c2b29ed6977dc4912/eclipse-platform-parent/pom.xml#L478).

Therefore `format` has already a value assigned and the property `tycho.buildqualifier.format`, which is new in Tycho 3.0.0 is not considered while populating the plugin's property values.

I created PR https://github.com/eclipse-platform/eclipse.platform/pull/61 to fix that but I think the cleaner approach would be to use `tycho.buildqualifier.format` in the platform-parent pom instead, to allow lower level overwrites.

> 
> Maven build **replaces** built version with cached version of **org.eclipse.sdk_4.25.0.v20220718-1722** bundle now! https://ci.eclipse.org/releng/view/Builds/job/I-build-4.25/100/consoleText

This happened because of the unchanged build-qualifier. If the build-qualifier is the same like in the base-line the `Tycho-p2-plugin` replaces the jar build locally with the one from the base-line to ensure bit-by-bit identical build results.
